moon/cfc_77.15.580/sv_cmd.moon

```diff
@@ -0,0 +1,35 @@
+import concat from table
+import flaggedMessages,
+       cmdClearTime,
+       cmdShouldBan,
+       cmdSpamThreshold,
+       cmdExtremeSpamThreshold,
+       cmdSpam,
+       Logger,
+       Webhooker
+       from Section580
+
+originalRun = concommand.Run
+
+concommand.Run = (ply, cmd, args, argStr) ->
+    if not IsValid ply
+        return originalRun ply, cmd, args, argStr
+
+    if not cmd
+        return originalRun ply, cmd, args, argStr
+
+    plySteamID = "<Unknown Steam ID>"
+    plyNick = "<Unknown Player Name>"
+    plyIP = "<Unknown Player IP>"
+    plyIsValid = IsValid ply
+
+    if plyIsValid
+        plySteamID = ply\SteamID!
+        plyNick = ply\Nick!
+        plyIP = ply\IPAddress!
+
+    plyCmdSpam = rawget cmdSpam, plySteamID
+    plyCmdSpam or= {}
+    plyCmdSpam[strName] or= 0
+    plyCmdSpam[strName] += 1
```

Comment:
Sadly this wont work if `cmdSpam[plySteamID]` is `nil`, as `plyCmdSpam` will become the new table, but that won't get assigned back into `cmdSpam[plySteamID]`
The most efficient way to do this (i think) would be to not use `or=`

something like
```moonscript
plyCmdSpam = rawget cmdSpam, plySteamID
if not plyCmdSpam
    plyCmdSpam = {}
    rawset cmdSpam, plySteamID, plyCmdSpam

plyCmdSpam[strName] or= 0
plyCmdSpam[strName] += 1